sycl/test/check_device_code/cuda/matrix/compile-query.cpp

```diff
@@ -0,0 +1,33 @@
+// REQUIRES: cuda
```

Comment:
Nit: should we update the name of the test to be "matrix-nvptx-compile-query-test.cpp" to follow other test names in this folder?
I'm actually not voting for it, because, I believe, "matrix", "nvptx" and "test" are redundant, as full path contains all necessary information to identify the test.